app/src/main/java/com/elavon/converge/processor/ConvergeClient.java

```diff
@@ -63,32 +63,41 @@ public void updateCredentials(
 
     public <T extends ElavonResponse> void call(final ElavonRequest model, final ConvergeCallback<T> callback) {
 
-        final Request request = getRequest(model);
-        final Callback cb = new Callback() {
-            @Override
-            public void onResponse(final Call call, final Response response) throws IOException {
-                if (response.isSuccessful()) {
-                    final T body;
-                    try {
-                        final String xml = response.body().string();
-                        final Class<T> clazz = ((Class) ((ParameterizedType) callback.getClass().getGenericSuperclass()).getActualTypeArguments()[0]);
-                        body = getResponse(xml, clazz);
-                    } catch (final Exception e) {
-                        callback.onFailure(e);
-                        return;
+        try {
+            final Request request = getRequest(model);
+            final Callback cb = new Callback() {
+                @Override
+                public void onResponse(final Call call, final Response response) throws IOException {
+                    if (response.isSuccessful()) {
+                        final T body;
+                        try {
+                            final String xml = response.body().string();
+                            final Class<T> clazz = ((Class) ((ParameterizedType) callback.getClass().getGenericSuperclass()).getActualTypeArguments()[0]);
+                            body = getResponse(xml, clazz);
+                        } catch (final Exception e) {
+                            callback.onFailure(e);
+                            return;
+                        }
+                        callback.onResponse(body);
+                    } else {
+                        callback.onFailure(new ConvergeClientException("Failed request with http status code: " + response.code()));
                     }
-                    callback.onResponse(body);
-                } else {
-                    callback.onFailure(new ConvergeClientException("Failed request with http status code: " + response.code()));
                 }
-            }
 
-            @Override
-            public void onFailure(final Call call, final IOException e) {
-                callback.onFailure(e);
+                @Override
+                public void onFailure(final Call call, final IOException e) {
+                    callback.onFailure(e);
+                }
+            };
+            if (request != null) {
+                client.newCall(request).enqueue(cb);
+            } else {
+                callback.onFailure(new ConvergeClientException("Insufficient authorization for transaction."));
```

Comment:
wait is this needed with the change below to throw exception ? would we ever have null request ?